src/Catty/CloneBrick+Instruction.swift

```diff
@@ -0,0 +1,900 @@
+                    self.parseBrick(script: newWhenScript, oldScript: whenScript, variables: object.userData.variables(), lists: object.userData.lists())
+                    object.scriptList.add(newWhenScript)
+
+                case let whenTouchDownScript as WhenTouchDownScript:
+                    let newWhenTouchDownScript = WhenTouchDownScript()
+                    newWhenTouchDownScript.object = object
+                    self.parseBrick(script: newWhenTouchDownScript, oldScript: whenTouchDownScript, variables: object.userData.variables(), lists: object.userData.lists())
+                    object.scriptList.add(newWhenTouchDownScript)
+
+                case let whenBackgroundChangesScript as WhenBackgroundChangesScript:
+                    let newWhenBackgroundChangesScript = WhenBackgroundChangesScript()
+                    newWhenBackgroundChangesScript.object = object
+                    self.parseBrick(script: newWhenBackgroundChangesScript, oldScript: whenBackgroundChangesScript, variables: object.userData.variables(), lists: object.userData.lists())
+                    object.scriptList.add(newWhenBackgroundChangesScript)
+
+                case let broadcastScript as BroadcastScript:
+                    let newBroadcastScript = BroadcastScript()
+                    newBroadcastScript.object = object
+                    self.parseBrick(script: newBroadcastScript, oldScript: broadcastScript, variables: object.userData.variables(), lists: object.userData.lists())
+                    object.scriptList.add(newBroadcastScript)
+
+                case let whenConditionScript as WhenConditionScript:
+                    let newWhenConditionScript = WhenConditionScript()
+                    newWhenConditionScript.object = object
+                    self.parseBrick(script: newWhenConditionScript, oldScript: whenConditionScript, variables: object.userData.variables(), lists: object.userData.lists())
+                    object.scriptList.add(newWhenConditionScript)
+
+                default:
+                    break
+                }
+            }
+
+            object.soundList = objectToClone.soundList.mutableCopy() as? NSMutableArray
+            object.lookList = objectToClone.lookList.mutableCopy() as? NSMutableArray
+
+            objectToClone.scene.add(object: object)
+            let spriteNode = CBSpriteNode(spriteObject: object)
+            object.spriteNode = spriteNode
+
+            let stage = objectToClone.spriteNode.scene as? Stage
+            stage?.addChild(object.spriteNode)
+            object.spriteNode.startAsCopy(objectToClone.spriteNode)
+            stage?.scheduler.registerSpriteNode(object.spriteNode)
+            stage?.addCopiedObjecToProject(spriteObject: object)
+        }
+    }
+
+    func parseBrick(script: Script, oldScript: Script, variables: [UserVariable], lists: [UserList]) {
+
+        for brick in oldScript.brickList {
+
+            switch brick {
```

Comment:
We should really make the Bricks, Scripts and Variables conform to `NSCopying` and then just call `.copy()` on generic objects instead of making gigantic switch cases. It's complicated to understand and this file would have to be edited for every new brick in the future.

Reply:
I created a "clone" function instead of using the copy functions. Because I pass the script as parameter and so I can set the member variables for the bricks. Some bricks need uservariable, userlist or other bricks and with the script I can set all these member var.